src/frameworks/next/index.ts

```diff
@@ -210,10 +246,37 @@ export async function ɵcodegenPublicDirectory(sourceDir: string, destDir: strin
       }
     }
 
-    const prerenderManifestBuffer = await readFile(
-      join(sourceDir, distDir, "prerender-manifest.json")
-    );
+    const [prerenderManifestBuffer, routesManifestBuffer] = await Promise.all([
+      readFile(
+        join(
+          sourceDir,
+          distDir,
+          "prerender-manifest.json" // TODO: get this from next/constants
+        )
+      ),
+      readFile(
+        join(
+          sourceDir,
+          distDir,
+          "routes-manifest.json" // TODO: get this from next/constants
+        )
+      ),
+    ]);
+
     const prerenderManifest = JSON.parse(prerenderManifestBuffer.toString());
+    const routesManifest = JSON.parse(routesManifestBuffer.toString()) as Manifest;
+
+    const rewritesToUse = getNextjsRewritesToUse(routesManifest.rewrites);
+    const rewritesNotSupportedByFirebase = rewritesToUse?.filter(
+      (rewrite) => isRewriteSupportedByFirebase(rewrite) === false
+    );
+    const redirectsNotSupportedByFirebase = routesManifest.redirects?.filter(
```

Comment:
Can we destructure with defaults, `const { redirects=[] } = routesManifest` like we do in build? would make it so this couldn't be null

Reply:
and we wouldn't have to guard